display_list/dl_op_receiver.h

```diff
@@ -23,11 +23,19 @@ namespace flutter {
 class DisplayList;
 
 //------------------------------------------------------------------------------
-/// @brief      The pure virtual interface for interacting with a display list.
-///             This interface represents the methods used to build a list
-///             through the DisplayListBuilder and also the methods that will be
-///             invoked through the DisplayList::dispatch() method.
+/// @brief      Internal API for rendering recorded display lists to backends.
 ///
+/// The |DisplayList| object will play back recorded operations in this format.
+/// Most developers should not needed to deal with this interface unless they
+/// are writing a utility that needs to examine the contents of a display list.
+///
+/// Similar to |DlCanvas|, the only state carried by implementations of this
+/// interface are the clip and transform which are saved and restored by the
+/// |save|, |saveLayer|, and |restore| calls.
```

Comment:
Unlike DlCanvas, this interface has attribute state which is global across an entire DisplayList (not affected by save/restore).

Reply:
Basically at the start of processing any given DL, the attributes should all have default state and then modified each time the receiver gets a setAttribute() call.

[Side note: This gets tricky for the renderers that implement this. Utilities that are simply scanning a DL can just skip over the `drawDisplayList` calls, but renderers have to immediately process those DisplayLists in situ and so they have to maintain separate attribute states for the current DL and the new DL that they recurse into. For Skia I went with just creating a brand new dispatcher and feeding the new DL into that. For Impeller, I reuse "this" and just do a lot of saving state and restoring of state before/after...]

Reply:
Done!

Reply:
You should remove the "the only state" as it isn't the only state. Something like "similar to ... this interface carries clip and transform state which ..."

Reply:
Done!